examples/common/optim/outlier_detection.py

```diff
@@ -0,0 +1,32 @@
+
+import collections
+
+class OutlierDetector:
+    
+    def __init__(self,
+                 threshold=7.5,
+                 increase_lookback=500):
+          
+        self.intermediate_data_queue = collections.deque(maxlen=increase_lookback)
+        self.slow_moving_average = collections.deque(maxlen=increase_lookback)
+        self.outlier_threshold = threshold
+    
+    def insert_observation(self, obs):
```

Comment:
would be good to add a docstring here explaining that this function both inserts the item into the tracking queues, and returns whether the item is considered an outlier. Might also change the function name to reflect this. Or, if reasonable, might be nicer to break up into two functions, one which checks if an observation is an outlier, and one which inserts the observation into the tracking queues.